cluster/src/test/java/com/hartwig/pipeline/smoke/SmokeTest.java

```diff
@@ -59,48 +58,67 @@ public void tearDown() throws Exception {
+        final String fixtureDir = "smoke_test/" + inputMode + "/";
+        final ImmutableArguments.Builder builder = Arguments.defaultsBuilder(Arguments.DefaultsProfile.DEVELOPMENT.toString())
+                .sampleJson(Resources.testResource(fixtureDir + "samples.json"))
                 .cloudSdkPath(CLOUD_SDK_PATH)
-                .setId(setName)
-                .runId(runId)
+                .setId(inputMode)
+                .runId(version)
                 .runGermlineCaller(false)
-                .sbpApiRunId(SBP_RUN_ID)
-                .sbpApiUrl(apiUrl)
-                .rclonePath(RCLONE_PATH)
-                .rcloneGcpRemote(GCP_REMOTE)
-                .rcloneS3RemoteDownload("s3")
                 .cleanup(true)
-                .outputBucket("services-pipeline-output-pilot-1")
-                .context(Context.SERVICES)
-                .build();
+                .outputBucket("smoketest-pipeline-output-pilot-1")
+                .context(Context.DIAGNOSTIC);
+
+        final String username = System.getProperty("user.name");
+        if (username.equals("root")) {
+            String privateKeyPath = workingDir() + "/google-key.json";
+            builder.privateKeyPath(privateKeyPath).uploadPrivateKeyPath(privateKeyPath);
+        } else {
+            builder.cloudSdkPath(String.format("/Users/%s/google-cloud-sdk/bin", username));
+        }
+
+        Arguments arguments = builder.build();
         Storage storage = StorageProvider.from(arguments, CredentialProvider.from(arguments).get()).get();
 
-        cleanupBucket(setName, arguments.outputBucket(), storage);
+        cleanupBucket(inputMode, arguments.outputBucket(), storage);
 
         PipelineState state = victim.start(arguments);
-        assertThat(state.status()).isEqualTo(PipelineStatus.QC_FAILED);
+        assertThat(state.status()).isEqualTo(expectedStatus);
 
-        File expectedFilesResource = new File(Resources.testResource("smoke_test/expected_output_files"));
+        File expectedFilesResource = new File(Resources.testResource(fixtureDir + "expected_output_files"));
         List<String> expectedFiles = FileUtils.readLines(expectedFilesResource, FILE_ENCODING);
         List<String> actualFiles = listOutput(setName, arguments.outputBucket(), storage);
         assertThat(actualFiles).containsOnlyElementsOf(expectedFiles);
-
-        assertThatAlignmentIsEqualToExpected(setName, REFERENCE_SAMPLE, arguments.outputBucket(), storage);
-        assertThatAlignmentIsEqualToExpected(setName, TUMOR_SAMPLE, arguments.outputBucket(), storage);
-
         cleanupBucket(setName, arguments.outputBucket(), storage);
     }
 
+    @NotNull
+    private String version(final String inputMode) {
+        String version = System.getProperty("version");
+        if (version.equals("local-SNAPSHOT")) {
+            version = System.getProperty("user.name") + "-" + inputMode;
+        }
+        return version.length() > 14 ? version.substring(0, 14) : version;
```

Comment:
This is a bit cryptic, I assume the `14` refers to the length of the string `local-SNAPSHOT`?

Reply:
Yes, i'll extract a constant.